pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java

```diff
@@ -543,9 +543,10 @@ private static String getErrorMessage(HttpUriRequest request, CloseableHttpRespo
     StatusLine statusLine = response.getStatusLine();
     String reason;
     try {
-      reason = JsonUtils.stringToJsonNode(EntityUtils.toString(response.getEntity())).get("_error").asText();
+      String entityStr = EntityUtils.toString(response.getEntity());
+      reason = JsonUtils.stringToObject(entityStr, SimpleHttpErrorInfo.class).getError();
```

Comment:
Have we considered backward compatibility here since the server and client may not use the same version at all times (e.g. during deployment)?

Reply:
The incompatibility issue you're referring to comes into picture when/if the object definition is changed. This PR only refactors the way the POJO is deserialized. It does not change the properties of object itself.

Reply:
I understand that we didn't change the properties in this PR. The concern I have here is that client code and server code may not run the same version. Since the error message can come from either `error` or `_error`, should we consider the backward compatibility here, like trying to fetch both fields?

One scenario I can think of is that if the response still shows `_error` in the payload, when the `stringToObject()` method in Line 547 gets invoked, it will still fail because no `_error` is specified as a JsonProperty. In this case, we still cannot parse the payload correctly.

Reply:
This is a valid concern and we'll have incompatibility between different versions for different component. However, as I mentioned before, this PR is just a refactoring and the original backward incompatible changes in the data definition  shouldn't have been allowed in the first place. Since this is not in the success path and we'll know about the error by catching the exception, I believe it makes the code simpler and more maintainable if we don't have a special code handling "error", "_error", and any future names.